### PR TITLE
Move the AzConfig SDK to the central repo

### PR DESCRIPTION
**Pull request by AlexGhiondea**
Move the code for the AzConfig service into the central repo.

During the move I added a P2P reference to the Azure.Core package. That will be changed once the package name for Azure.Core is final.

/cc @weshaggard @KrzysztofCwalina @maririos @Petermarcu

### CODE REVIEW

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs
Comment (maririos) [inline src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: We should disable running these test in the YAML definition and not in the code. Otherwise, users that want to run this code will need to manually enable the test again.
The way we did it in the lab repo was like this: https://github.com/Azure/azure-sdk-for-net-lab/blob/master/.azure-pipelines/azure-pipelines.yml#L43

Comment (AlexGhiondea) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: Is that how we want to do it in the central repo as well? I am fine with that, @weshaggard what do you think?

Even so, I want to keep the tests disabled until we figure out that part (ie not block this PR on that) — there is an issue tracking the work to reenable them and it is high on the priority list for me.

Comment (AlexGhiondea) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: I would also like to move to XUnit once we have a plan for how to run different categories of tests in CI vs not so that we have a single way of running tests.

Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: So there are 3 things here:
- Figure out how to handle CI for live test. cc: @mikeharder => I still consider it is better to do it on CI and not in the .cs files.
- Change the tests to XUnit? Is this to be consistent with the rest of the tests in this repository?
- Fix the tests to reflect the changes the service did to their REST API. The culprit is PR https://msazure.visualstudio.com/Azure%20AppConfig/_git/AppConfigService/commit/5120219050524e3e64d55eaa0512c964f0a3f5d2?refName=refs%2Fheads%2Fdev
We had issues tracking this, but because we didn't get notified, we were not ready for the changes.
Issues tracking the work are: 
https://github.com/Azure/azure-sdk-for-net-lab/issues/101
https://github.com/Azure/azure-sdk-for-net-lab/issues/99
https://github.com/Azure/azure-sdk-for-net-lab/issues/112

Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: I will work on bullet point 3, so our tests, reflect the changes in the service

Comment (AlexGhiondea) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: @maririos I agree and I think those things can happen after this PR is merged in.

The XUnit move is so that we are consistent with the repo which will make it that filtering live tests will be easier to achieve in a common way.

Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: FYI @lenala breacking changes happened to the service.

Comment (mikeharder) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: NUnit vs xUnit is an interesting question.  I personally prefer NUnit for a few reasons:

1. NUnit has more granular control over test parallelization.  xUnit only allows running test **classes** in parallel, but not test **methods** within a class.  NUnit uses a hierarchy of attributes which can be applied to assemblies, classes, and methods to enable or disable parallelism at any level.  If you have a large test suite with different parallelism requirements (which I suspect we will), NUnit is a much better fit.

2. When using parameterized tests, xUnit requires all parameters to be serializable in order for each parameterized test to appear in the VS test runner.  If any parameters are not serializable, VS will show all the parameterized tests as a single test case.  NUnit does not have this restriction -- as long as the test parameters have unique `ToString()` representations, VS can show and run each parameterized test.

So my personal preference would be to standardize on NUnit and move existing tests from xUnit to NUnit over time, perhaps when they are converted to support parallelism, dynamically creating test resources, etc.

At a minimum, I would **not** convert these existing tests from NUnit to xUnit, until we have finalized our test framework decision.

Comment (AlexGhiondea) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: Sounds good — I don’t have a preference either way — I thought we were using xUnit.

The test conversion is something that can be done later.

Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: Nice! I'm almost done with the changes to the live tests. 
@bsiegel do you have insight into bullet point 1? I think that is the remaining part in order for us to merge this PR.

Comment (bsiegel) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: @maririos Right now we aren't running live tests as part of CI. This is something we want to enable down the road. I know we have a standard framework for recording live traffic, and playing back test runs either against those recordings or against live Azure, but I'm not sure under what conditions we're currently using the recordings vs. the live tests. I can track down some more info for you on this - are you of the opinion that live tests for CI builds is a must-have for this service?

Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: >are you of the opinion that live tests for CI builds is a must-have for this service?

No, I am not. 
See our YAML definition for the lab repo: https://github.com/Azure/azure-sdk-for-net-lab/blob/master/.azure-pipelines/azure-pipelines.yml#L43
I would want something like that, where the logic of what tests should run is in the YAML/build system and not in the .cs files.

Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: Public build runs the mock tests, internal build runs all tests (mocks and live ones)

Comment (AlexGhiondea) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: @bsiegel @weshaggard @maririos if you don't have any objections with the current state of this PR I am going to merge this in -- we can handle all the remaining issues after this PR is merged.

Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration.Tests/ConfigurationLiveTests.cs]: :shipit:

**Review by mikeharder — APPROVED** [review]

**Pull request merged**
